tests/index.ts

```diff
@@ -111,4 +119,20 @@ export class ProtocolParserUtils {
     expect(actualAction.sender).toBe(expectedAction.sender);
     expect(actualAction.recipient).toBe(expectedAction.recipient);
   }
+
+  public assertSingleSwapAction(expectedAction: ISingleSwapAction, actualAction: ISingleSwapAction): void {
+    expect(actualAction.fromToken).toBe(expectedAction.fromToken);
+    expect(actualAction.toToken).toBe(expectedAction.toToken);
+    expect(actualAction.fromAmount).toBe(expectedAction.fromAmount);
+    expect(actualAction.toAmount).toBe(expectedAction.toAmount);
+    expect(actualAction.recipient).toBe(expectedAction.recipient);
+  }
+
+  public assertMultiSwapAction(expectedAction: IMultiSwapAction, actualAction: IMultiSwapAction): void {
+    expect(actualAction.fromTokens).toEqual(expectedAction.fromTokens);
+    expect(actualAction.toTokens).toEqual(expectedAction.toTokens);
+    expect(actualAction.fromAmounts).toEqual(expectedAction.fromAmounts);
+    expect(actualAction.toAmounts).toEqual(expectedAction.toAmounts);
+    expect(actualAction.recipients).toEqual(expectedAction.recipients);
```

Comment:
recipient is an array, this wont work. 
You need to loop over the items and assert each one with another

Reply:
Same for the items above where the item is an array